lib/plug/router.ex

```diff
@@ -334,21 +334,42 @@ defmodule Plug.Router do
         true ->
           raise ArgumentError, message: "expected :do to be given as option"
       end
-
+  
     {path, guards} = extract_path_and_guards(expr)
-
+    private_options = extract_private_options(options)
+    
     quote bind_quoted: [method: method,
                         path: path,
                         options: options,
+                        private_options: Macro.escape(private_options, unquote: true),
                         guards: Macro.escape(guards, unquote: true),
                         body: Macro.escape(body, unquote: true)] do
+
       {method, match, host, guards} = Plug.Router.__route__(method, path, guards, options)
+
       defp do_match(unquote(method), unquote(match), unquote(host)) when unquote(guards) do
-        fn var!(conn) -> unquote(body) end
+        fn conn ->
+          var!(conn) = %{ conn | private: Map.merge(conn.private, unquote(private_options)) }
+          unquote(body)
+        end
       end
     end
   end
 
+  # Pops key :private from options and checks if it is a Map. Case it is not, 
+  # raises early so that the developer knows in compilation time. Currently it
+  # does not accept options from forward calls but does not raise.
+  defp extract_private_options([]), do: {:%{}, [], []}
+  defp extract_private_options({:options, _, _}), do: {:%{}, [], []} 
```

Comment:
Okay! Will follow the __route__ route ( :+1:). It seemed like a bug to me but I didn't want to change the forward macro in this same PR. I added the comment and a test for forwarding with options because of that. If you think that is better doing it all here I will be glad to take a look in it.